drv/lpc55-spi-server/src/main.rs

```diff
@@ -8,24 +8,329 @@
 #![no_std]
 #![no_main]
 
+use drv_lpc55_gpio_api::{Direction, Pin, Value};
 use drv_lpc55_spi as spi_core;
 use drv_lpc55_syscon_api::{Peripheral, Syscon};
+use drv_spi_msg::*;
 use lpc55_pac as device;
 use ringbuf::*;
+
+use crc::{Crc, CRC_32_CKSUM};
+use lpc55_romapi::bootrom;
 use userlib::*;
 
 task_slot!(SYSCON, syscon_driver);
 task_slot!(GPIO, gpio_driver);
 
+// See drv/spi-msg/README.md
```

Comment:
I'm not at peace with the naming yet.
I'm thinking that on the SP side, there should be a single task for the SP:spi4 driver that incorporates the low level SPI and the protocol instead of a spi server and a spi-rot server.
The UART to SP task might be able to be merged as well, so one task would own a SPI and UART interface as well as an Idl interface. I'm not ready for that level of complexity yet if that were actually a good idea.

Reply:
Hmmm, do we have a name for the "SP to RoT" channel?

I don't like using `2` for "to" – how does `{stm32h7,lpc55}-spi-rot-server` sound?  It's a bit shorter and still pretty unambiguous, since on both sides the server is handling RoT-related SPI communication.